src/AppBundle/Helper/CardFilter/CardFilterFactory.php

```diff
@@ -103,4 +95,17 @@ public function getFilterClasses()
     {
         return $this->filterClasses;
     }
-}
\ No newline at end of file
+
+    private function getFilesInDirectoryAndSubdirectories (string $path = __DIR__)
+    {
+        $arrayOfClasses = [];
+        foreach (array_keys(\Composer\Autoload\ClassMapGenerator::createMap($path)) as $class) {
+            if ($class === AbstractCardFilter::class || $class === CardFilterInterface::class || $class === __CLASS__) {
```

Comment:
Is it possible to simplify this into using `class_implements()`  like you've done in the `validateFilterClasses()` function and then use the `isAbstract()` function from ReflectionClass (https://www.php.net/manual/en/reflectionclass.isabstract.php) to help futureproof us for if additional abstract classes are added?